---
title: "initialAnalysis"
output:
  pdf_document: default
  html_document: default
---

```{r setup, include=FALSE}
# #reference genome 
# library(BSgenome.Mmusculus.UCSC.mm10)
# 
# #annotations
# library(org.Mm.eg.db)

#creating count matrix
library(tximeta)
library(BiocManager)
library(dplyr)
library(DESeq2)
```

## Protocol:

### 1. Use salmon to quantify transcripts
### 2. Use tximeta to get count matrix
### 3. use with DEseq for expression


import data table & create table with file paths to sample quant files
``` {r , include = TRUE, echo=FALSE}

data_table = read.delim("data_table")

txdata = data_table %>% dplyr::select(c(SampleName, SampleGenotype, SampleTime, QuantFile)) %>%
  dplyr::rename(names = SampleName, files = QuantFile)

dirPath = "/active/allenspach_e/AllenspachRNASeqData/2022_SH2B3_TcellSeq/data/"

#rename files column to include full path

txdata = txdata %>% mutate(files = paste(dirPath, files, sep = ""))

#import quantifications
se = tximeta(data.frame(as_tibble(txdata)))

#get quants at gene level
gse = summarizeToGene(se)


# analyzedPath = "/active/allenspach_e/AllenspachRNASeqData/2022_SH2B3_TcellSeq/analyzed_data/"
# 
# #save datasets as RDS
# saveRDS(gse, paste(analyzedPath,"gse.rds", sep = ""))
# saveRDS(se, paste(analyzedPath,"se.rds", sep = ""))

```


```{r createCountMatrix, include = T, echo=FALSE}

#differential expression with DESeq2
dds = DESeqDataSet(gse, design = ~ SampleGenotype + SampleTime)

#filter out empty rows

#get empty rows
nonzero = rowSums(counts(dds)) > 1
dds = dds[nonzero, ]

#factor so genotypes are grouped together
dds$SampleGenotype = factor(dds$SampleGenotype, levels = c("WT", "KO")) %>%
  relevel(dds$SampleGenotype, ref = "WT")

```

```{r diffexp, include = T}

dds = DESeq(dds)

#log2 fold changes and pvalues for WT vs KO
dds.res = results(dds, contrast = c("SampleGenotype", "WT", "KO"))

```


```{r annotate_dds, include = F}
library(AnnotationDbi)
library(org.Mm.eg.db)

#map ids from logfold change (shrinked)
dds.res$symbol <- mapIds(org.Mm.eg.db,
                    keys = row.names(dds.res),
                    column = "SYMBOL",
                    keytype = "ENSEMBL",
                    multiVals = "first")

```



```{r distMatrix, echo=FALSE}

library(PoiClaClu)
library(RColorBrewer)
poisd = PoissonDistance(t(counts(dds)))


samplePoisDistMatrix = as.matrix(poisd$dd)

colors <- colorRampPalette((brewer.pal(9, "Blues")) )(255)
rownames(samplePoisDistMatrix) = paste(dds$SampleGenotype, dds$SampleTime, sep = " - ")
colnames(samplePoisDistMatrix) = NULL
pheatmap::pheatmap(samplePoisDistMatrix, clustering_distance_rows = poisd$dd,
         clustering_distance_cols = poisd$dd, col = colors)
```


```{r PCAplots, include = T}

library(vsn)

#variance stabilizing transformation
vsd = vst(dds)
plotPCA(vsd, intgroup = c("SampleGenotype", "SampleTime"))
```
```{r logfoldchange, include = F}
library(apeglm)

#different comparisons we can use
resultsNames(dds)

```

```{r genecluster, include = T}

library(genefilter)


topVarGenes = head(order(rowVars(assay(vsd)), decreasing = T), 20)
mat = assay(vsd)[ topVarGenes, ]

rownames(mat) <- mapIds(org.Mm.eg.db,
                    keys = rownames(mat),
                    column = "SYMBOL",
                    keytype = "ENSEMBL",
                    multiVals = "first")

mat = mat - rowMeans(mat)
anno = as.data.frame(colData(vsd)[, c("SampleGenotype", "SampleTime")])

pheatmap::pheatmap(mat, annotation_col = anno)
```



```{r logfoldchange_shrink, include = T}

library(apeglm)

resultsNames(dds)

#calculate lfc shrink for 0hr
res = lfcShrink(dds, coef = "SampleGenotype_KO_vs_WT", type = "apeglm")

plotMA(res, ylim = c(-5, 5))
```


```{r deseq_timecourse, include = T}

ddsTC = DESeqDataSet(gse, design = ~ SampleGenotype + SampleTime + SampleGenotype:SampleTime)

#likelyhood ratio test
ddsTC = DESeq(ddsTC, test="LRT", reduced = ~ SampleGenotype + SampleTime)

resTC = results(ddsTC)
resTC$symbol = mcols(ddsTC)$symbol

```
