---
title: "initialAnalysis"
output: html_document
---

```{r setup, include=FALSE}
# #reference genome 
# library(BSgenome.Mmusculus.UCSC.mm10)
# 
# #annotations
# library(org.Mm.eg.db)

#creating count matrix
library(tximeta)
library(BiocManager)
library(dplyr)
library(DESeq2)
```

## Protocol:

### 1. Use salmon to quantify transcripts
### 2. Use tximeta to get count matrix
### 3. use with DEseq for expression


import data table & create table with file paths to sample quant files
``` {r include = TRUE}

data_table = read.delim("data_table", stringsAsFactors = F)

txdata = data_table %>% dplyr::select(c(SampleName, SampleGenotype, SampleTime, QuantFile)) %>%
  dplyr::rename(names = SampleName, files = QuantFile)

dirPath = "/active/allenspach_e/AllenspachRNASeqData/2022_SH2B3_TcellSeq/data/"

#rename files column to include full path

txdata = txdata %>% mutate(files = paste(dirPath, files, sep = ""))

#import quantifications
se = tximeta(data.frame(as_tibble(txdata)))

#get quants at gene level
gse = summarizeToGene(se)

```



