---
title: "timecourse_Analysis"
output: pdf_document
---
---
title: "RNAseq: CD8+ OT-I Stimulation Timecourse in SH2B3 mice"
output:
  pdf_document: default
  html_document: default
---

```{r setup, include=FALSE}
# #reference genome 
# library(BSgenome.Mmusculus.UCSC.mm10)
# 
# #annotations
# library(org.Mm.eg.db)

#creating count matrix
library(tximeta)
library(BiocManager)
library(dplyr)
library(DESeq2)
```

## Protocol:

### 1. Use salmon to quantify transcripts
### 2. Use tximeta to get count matrix
### 3. use with DEseq for expression


import data table & create table with file paths to sample quant files
``` {r , echo = F}

data_table = read.delim("data_table")

txdata = data_table %>% dplyr::select(c(SampleName, SampleGenotype, SampleTime, QuantFile)) %>%
  dplyr::rename(names = SampleName, files = QuantFile)

dirPath = "/active/allenspach_e/AllenspachRNASeqData/2022_SH2B3_TcellSeq/data/"

#rename files column to include full path

txdata = txdata %>% mutate(files = paste(dirPath, files, sep = ""))

#import quantifications with automatic metadata
se = tximeta(data.frame(as_tibble(txdata)))

#get quants at gene level (removing here)
gse = summarizeToGene(se)

```

Using DESeq to get read counts

```{r createCountMatrix, include = T}

#differential expression with DESeq2
dds = DESeqDataSet(gse, design = ~ SampleGenotype + SampleTime + SampleGenotype:SampleTime)

#use if want to do transcript specific
# dds = DESeqDataSet(se, design = ~ SampleGenotype + SampleTime + SampleGenotype:SampleTime)

#filter out empty rows

#get empty rows
nonzero = rowSums(counts(dds)) > 1
dds = dds[nonzero, ]

#factor so genotypes are grouped together
dds$SampleGenotype = factor(dds$SampleGenotype, levels = c("WT", "KO")) %>%
  relevel(dds$SampleGenotype, ref = "WT")

```

```{r diffexp, include = T}

dds = DESeq(dds)

#log2 fold changes and pvalues for WT vs KO
dds.res = results(dds, contrast = c("SampleGenotype", "WT", "KO"))

```

Annotating dataframe with gene symbols

```{r annotate_dds, include = F}
library(AnnotationDbi)
library(org.Mm.eg.db)

#map ids from logfold change (shrinked)


dds.res$symbol = mapIds(org.Mm.eg.db,
                         keys = rownames(dds.res),
                         keytype = "ENSEMBL",
                         column = "SYMBOL")



#use if want transcript specific keys
# dds.res$symbol <- mapIds(org.Mm.eg.db,
#                     keys = rownames(dds.res),
#                     column = "SYMBOL",
#                     keytype = "ENSEMBLTRANS",
#                     multiVals = "first")

```


Visualizing sample distances
```{r distMatrix, echo=FALSE}

library(PoiClaClu)
library(RColorBrewer)
poisd = PoissonDistance(t(counts(dds)))


samplePoisDistMatrix = as.matrix(poisd$dd)

colors <- colorRampPalette((brewer.pal(9, "Blues")) )(255)
rownames(samplePoisDistMatrix) = paste(dds$SampleGenotype, dds$SampleTime, sep = " - ")

#can remove column names but will keep here
colnames(samplePoisDistMatrix) = paste(dds$SampleGenotype, dds$SampleTime, sep = " - ")
pheatmap::pheatmap(samplePoisDistMatrix, clustering_distance_rows = poisd$dd,
         clustering_distance_cols = poisd$dd, col = colors)
```

PCA plot (all time points)

```{r PCAplots, include = T}

library(vsn)

#variance stabilizing transformation on non-normalized counts
vsd = vst(dds, blind = T)

#use ntop to get all rows of matrix (default is 500)
plotPCA(vsd, intgroup = c("SampleGenotype", "SampleTime"), ntop = nrow(vsd))
```

Gene cluster heatmap
```{r genecluster, include = T}

library(genefilter)


topVarGenes = head(order(rowVars(assay(vsd)), decreasing = T), 50)
mat = assay(vsd)[ topVarGenes, ]



rownames(mat) = mapIds(org.Mm.eg.db,
                    keys = rownames(mat),
                    column = "SYMBOL",
                    keytype = "ENSEMBL",
                    multiVals = "first")

# which genes are NA?
unmatched_ENSEMBL = rownames(mat)[which(is.na(rownames(mat)), arr.ind = F)]

#look like pseudogenes; remove
mat = mat[-which(is.na(rownames(mat))),]

mat = mat - rowMeans(mat)
anno = as.data.frame(colData(vsd)[, c("SampleGenotype", "SampleTime")])

# png(filename = "h1eatmap.png", width = 6, height = 6, units = "in", res = 300, type = "cairo")
pheatmap::pheatmap(mat, annotation_col = anno)
# while (!is.null(dev.list()))  dev.off()
```



```{r logfoldchange_shrink, include = F}

library(apeglm)

resultsNames(dds)

#calculate lfc shrink for 0hr
res = lfcShrink(dds, coef = "SampleGenotype_KO_vs_WT", type = "apeglm")

plotMA(res, ylim = c(-5, 5))
```

Processing data as time course:
Start with likelihood ratio test to remove genotype specific differences - 
the remaining genes with small p values showed genotype specific effects after
time 0hr.
```{r deseq_timecourse, include = T}
library(tidyverse)
ddsTC = DESeqDataSet(gse, design = ~ SampleGenotype + SampleTime + SampleGenotype:SampleTime)

#likelyhood ratio test
ddsTC = DESeq(ddsTC, test="LRT", reduced = ~ SampleGenotype + SampleTime)

resTC = results(ddsTC)
resTC$symbol = mcols(ddsTC)$symbol

stim = plotCounts(ddsTC, which.min(resTC$padj),
                  intgroup = c("SampleGenotype", "SampleTime"), returnData = T)
stim$hour = as.numeric(as.character(stim$SampleTime))

ggplot(stim,
       aes(x = SampleTime, y = count, color = SampleGenotype, group = SampleGenotype)) + geom_point() +
  stat_summary(fun = mean, geom = "line") + scale_y_log10()
```

Cluster significant genes by profile


```{r, echo = F}
betas = coef(ddsTC)

#plot top 20 genes
topGenes = head(order(resTC$padj), 20)
mat = betas[topGenes, -c(1,2)]

rownames(mat) <- mapIds(org.Mm.eg.db,
                    keys = rownames(mat),
                    column = "SYMBOL",
                    keytype = "ENSEMBL",
                    multiVals = "first")

thr = 3
mat[mat < -thr] = -thr
mat[mat > thr] = thr


pheatmap::pheatmap(mat, breaks = seq(from = -thr, to = thr, length = 101),
         cluster_col = F)
```





